flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/NonBufferResponseDecoder.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.io.network.netty;
+
+import org.apache.flink.shaded.netty4.io.netty.buffer.ByteBuf;
+import org.apache.flink.shaded.netty4.io.netty.channel.ChannelHandlerContext;
+
+import java.net.ProtocolException;
+
+import static org.apache.flink.runtime.io.network.netty.NettyMessage.BufferResponse;
+import static org.apache.flink.runtime.io.network.netty.NettyMessage.ErrorResponse;
+
+/**
+ * The decoder for messages other than {@link BufferResponse}.
+ */
+class NonBufferResponseDecoder extends NettyMessageDecoder {
+
+	/** The initial size of the message header accumulation buffer. */
+	private static final int INITIAL_MESSAGE_HEADER_BUFFER_LENGTH = 128;
+
+	/** The accumulation buffer of the message header. */
```

Comment:
Clarify whether this buffer is only for message header or whole message. If for message header, the below messageBuffer  -> messageHeaderBuf, otherwise we should remove `header` from the comment.

Reply:
Modified the comment since it should hold the whole buffer.